lib/signaling/v2/cancelableroomsignalingpromise.js

```diff
@@ -12,120 +12,133 @@ const {
         }
+        options.iceServers = iceServers;
+        peerConnectionManager.setConfiguration(options);
 
-        if (options.bandwidthProfile) {
-          transportOptions.bandwidthProfile = options.bandwidthProfile;
-        }
-
-        transportOptions = Object.assign({
-          automaticSubscription: options.automaticSubscription,
-          dominantSpeaker: options.dominantSpeaker,
-          environment: options.environment,
-          eventObserver: options.eventObserver,
-          logLevel: options.logLevel,
-          networkMonitor: options.networkMonitor,
-          networkQuality: options.networkQuality,
-          iceServerSourceStatus: iceServerSource.status,
-          insights: options.insights,
-          realm: options.realm,
-          sdpSemantics: options.sdpSemantics
-        }, transportOptions);
-
-        const Transport = options.Transport;
-        transport = new Transport(
-          options.name,
-          token,
-          localParticipant,
-          peerConnectionManager,
-          wsServer,
-          transportOptions);
-
-        transport.once('connected', function connected(initialState) {
+        return peerConnectionManager.createAndOffer().then(() => {
           if (isCanceled()) {
             reject(cancelationError);
-            return;
-          }
-          const localParticipantState = initialState.participant;
-          if (!localParticipantState) {
-            reject(new SignalingIncomingMessageInvalidError());
-            return;
+            throw cancelationError;
           }
+          // NOTE(mmalavalli): PeerConnectionManager#createAndOffer() queues the
+          // initial offer in the event queue for the 'description' event. So,
+          // we are dequeueing to prevent the spurious 'update' message sent by
+          // the client after connecting to a room.
+          peerConnectionManager.dequeue('description');
+        }, reject);
+      };
+
+      const {
+        InsightsPublisher,
+        NullInsightsPublisher,
+        automaticSubscription,
+        bandwidthProfile,
+        dominantSpeaker,
+        environment,
+        eventObserver,
+        logLevel,
+        name,
+        networkMonitor,
+        networkQuality,
+        insights,
+        realm,
+        sdpSemantics,
+        wsServerInsights
+      } = options;
+
+      const transportOptions = Object.assign({
+        automaticSubscription,
+        dominantSpeaker,
+        environment,
+        eventObserver,
+        logLevel,
+        networkMonitor,
+        networkQuality,
+        iceServers,
+        insights,
+        onIced,
+        realm,
+        sdpSemantics
+      }, typeof wsServerInsights === 'string' ? {
+        wsServerInsights
+      } : {}, InsightsPublisher ? {
+        InsightsPublisher
+      } : {}, NullInsightsPublisher ? {
+        NullInsightsPublisher
+      } : {}, bandwidthProfile ? {
+        bandwidthProfile
+      } : {});
+
+      transport = new Transport(
```

Comment:
I think we should move creation of transport earlier (beofre onIce is declared) - that would make this flow look better

Reply:
ah, never mind that's not possible since it takes onIce as a parameter.